src/handshake.rs

```diff
@@ -25,23 +25,27 @@ use tokio::io::AsyncRead;
 use tokio::io::AsyncWrite;
 
 use std::error::Error;
+use std::future::Future;
+use std::pin::Pin;
 
 use crate::Role;
 use crate::WebSocket;
 
 pub async fn client<S>(
-  request: Request<Body>,
+  executor: &dyn hyper::rt::Executor<Pin<Box<dyn Future<Output = ()>>>>,
```

Comment:
This might be better with a template arg:

```suggestion
pub async fn client<S, E>(
  executor: E,
  socket: S,
  request: Request<Body>,
) -> Result<(WebSocket<Upgraded>, Response<Body>), Box<dyn Error + Send + Sync>>
where
  S: AsyncRead + AsyncWrite + Send + Unpin + 'static,
  E: hyper::rt::Executor<Box<dyn Future<Output = ()> + Send + Unpin>> + 'static
{
  let (mut sender, conn) = hyper::client::conn::handshake(socket).await?;
  let fut = Box::pin(async move {
    if let Err(e) = conn.await {
      eprintln!("Error polling connection: {}", e);
    }
  });
  executor.execute(Box::new(fut));

```